NPE if https scheme is used by a UPnP device in the network

If certain Samsung devices are on the local network, jUPnP logs errors like

```
2018-09-16 23:05:03.295 [ERROR] [org.jupnp.transport.spi.StreamClient] - Request: HttpRequest[GET /capability HTTP/1.1]@7c95d877 failed
    java.lang.NullPointerException: Missing SslContextFactory
    at java.util.Objects.requireNonNull(Objects.java:228) ~[?:?]
    at org.eclipse.jetty.io.ssl.SslClientConnectionFactory.&lt;init&gt;(SslClientConnectionFactory.java:50) ~[74:org.eclipse.jetty.io:9.4.11.v20180605]
    at org.eclipse.jetty.client.HttpClient.newSslClientConnectionFactory(HttpClient.java:1155) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
    at org.eclipse.jetty.client.HttpDestination.newSslClientConnectionFactory(HttpDestination.java:138) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
    at org.eclipse.jetty.client.HttpDestination.&lt;init&gt;(HttpDestination.java:95) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
    at org.eclipse.jetty.client.PoolingHttpDestination.&lt;init&gt;(PoolingHttpDestination.java:25) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
    at org.eclipse.jetty.client.http.HttpDestinationOverHTTP.&lt;init&gt;(HttpDestinationOverHTTP.java:32) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
    at org.eclipse.jetty.client.http.HttpClientTransportOverHTTP.newHttpDestination(HttpClientTransportOverHTTP.java:51) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
    at org.eclipse.jetty.client.HttpClient.destinationFor(HttpClient.java:539) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
    at org.eclipse.jetty.client.HttpClient.send(HttpClient.java:572) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
    at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:727) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
    at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:680) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
    at org.jupnp.transport.impl.jetty.JettyStreamClientImpl$1.call(JettyStreamClientImpl.java:149) [210:org.jupnp:2.4.0]
    at org.jupnp.transport.impl.jetty.JettyStreamClientImpl$1.call(JettyStreamClientImpl.java:1) [210:org.jupnp:2.4.0]
    at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
    at java.lang.Thread.run(Thread.java:748) [?:?]
```
As it seems, jUPnP does not support https at all, but is expected to call a https url in this situation.
I haven't experienced any https in UPnP networks myself so far.
Even if the default SSL context were added to Jetty, I assume that requests would fail due to the impossibility to verify the certificate (which must be a self-signed one as there's no defined domain name for the device).
To circumvent this, we would probably have to add a custom trust manager, which simply accepts all certificates unseen.
If anybody has a better idea how to address this, please comment!

Enable debug for org.jupnp and got little more details:

```
2018-09-23 14:38:37.881 [DEBUG] [upnp.transport.spi.MulticastReceiver] - UDP datagram received from: 192.168.0.54:59718 on local interface: eth0 and address: 192.168.0.28
2018-09-23 14:38:37.882 [DEBUG] [org.jupnp.transport.Router          ] - Received asynchronous message: (IncomingDatagramMessage) NOTIFY
2018-09-23 14:38:37.883 [DEBUG] [p.protocol.RetrieveRemoteDescriptors] - Sending device descriptor retrieval message: (StreamRequestMessage) GET https://192.168.0.54:8888/capability
2018-09-23 14:38:37.884 [DEBUG] [org.jupnp.transport.Router          ] - Sending via TCP unicast stream: (StreamRequestMessage) GET https://192.168.0.54:8888/capability
2018-09-23 14:38:37.884 [ERROR] [org.jupnp.transport.spi.StreamClient] - Request: HttpRequest[GET /capability HTTP/1.1]@70d22028 failed
java.lang.NullPointerException: Missing SslContextFactory
	at java.util.Objects.requireNonNull(Objects.java:228) ~[?:?]
	at org.eclipse.jetty.io.ssl.SslClientConnectionFactory.<init>(SslClientConnectionFactory.java:50) ~[74:org.eclipse.jetty.io:9.4.11.v20180605]
	at org.eclipse.jetty.client.HttpClient.newSslClientConnectionFactory(HttpClient.java:1155) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
	at org.eclipse.jetty.client.HttpDestination.newSslClientConnectionFactory(HttpDestination.java:138) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
	at org.eclipse.jetty.client.HttpDestination.<init>(HttpDestination.java:95) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
	at org.eclipse.jetty.client.PoolingHttpDestination.<init>(PoolingHttpDestination.java:25) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
	at org.eclipse.jetty.client.http.HttpDestinationOverHTTP.<init>(HttpDestinationOverHTTP.java:32) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
	at org.eclipse.jetty.client.http.HttpClientTransportOverHTTP.newHttpDestination(HttpClientTransportOverHTTP.java:51) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
	at org.eclipse.jetty.client.HttpClient.destinationFor(HttpClient.java:539) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
	at org.eclipse.jetty.client.HttpClient.send(HttpClient.java:572) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:727) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:680) ~[70:org.eclipse.jetty.client:9.4.11.v20180605]
	at org.jupnp.transport.impl.jetty.JettyStreamClientImpl$1.call(JettyStreamClientImpl.java:149) [210:org.jupnp:2.4.0]
	at org.jupnp.transport.impl.jetty.JettyStreamClientImpl$1.call(JettyStreamClientImpl.java:1) [210:org.jupnp:2.4.0]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
2018-09-23 14:38:37.886 [WARN ] [org.jupnp.transport.spi.StreamClient] - HTTP request failed: (StreamRequestMessage) GET https://192.168.0.54:8888/capability
java.lang.NullPointerException: Missing SslContextFactory
	at java.util.Objects.requireNonNull(Objects.java:228) ~[?:?]
	at org.eclipse.jetty.io.ssl.SslClientConnectionFactory.<init>(SslClientConnectionFactory.java:50) ~[?:?]
	at org.eclipse.jetty.client.HttpClient.newSslClientConnectionFactory(HttpClient.java:1155) ~[?:?]
	at org.eclipse.jetty.client.HttpDestination.newSslClientConnectionFactory(HttpDestination.java:138) ~[?:?]
	at org.eclipse.jetty.client.HttpDestination.<init>(HttpDestination.java:95) ~[?:?]
	at org.eclipse.jetty.client.PoolingHttpDestination.<init>(PoolingHttpDestination.java:25) ~[?:?]
	at org.eclipse.jetty.client.http.HttpDestinationOverHTTP.<init>(HttpDestinationOverHTTP.java:32) ~[?:?]
	at org.eclipse.jetty.client.http.HttpClientTransportOverHTTP.newHttpDestination(HttpClientTransportOverHTTP.java:51) ~[?:?]
	at org.eclipse.jetty.client.HttpClient.destinationFor(HttpClient.java:539) ~[?:?]
	at org.eclipse.jetty.client.HttpClient.send(HttpClient.java:572) ~[?:?]
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:727) ~[?:?]
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:680) ~[?:?]
	at org.jupnp.transport.impl.jetty.JettyStreamClientImpl$1.call(JettyStreamClientImpl.java:149) ~[?:?]
	at org.jupnp.transport.impl.jetty.JettyStreamClientImpl$1.call(JettyStreamClientImpl.java:1) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
2018-09-23 14:38:37.897 [WARN ] [p.protocol.RetrieveRemoteDescriptors] - Device descriptor retrieval failed, no response: https://192.168.0.54:8888/capability
```

--- a/bundles/org.jupnp/src/main/java/org/jupnp/transport/spi/AbstractStreamClient.java
+++ b/bundles/org.jupnp/src/main/java/org/jupnp/transport/spi/AbstractStreamClient.java
@@ -26,6 +26,7 @@ import java.util.concurrent.TimeoutException;
 
 import org.jupnp.model.message.StreamRequestMessage;
 import org.jupnp.model.message.StreamResponseMessage;
+import org.jupnp.util.SpecificationViolationReporter;
 import org.jupnp.util.Exceptions;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -50,7 +51,7 @@ public abstract class AbstractStreamClient<C extends StreamClientConfiguration, 
         String protocol = split[0];
 
         if (protocol.equals("https")) {
-            log.debug("HTTPS invalid.  Ignoring call {}.",requestMessage.getUri());
+            SpecificationViolationReporter.report("HTTPS invalid.  Ignoring call " + requestMessage.getUri());
             return null;
         }
 	

--- a/bundles/org.jupnp/src/main/java/org/jupnp/transport/spi/AbstractStreamClient.java
+++ b/bundles/org.jupnp/src/main/java/org/jupnp/transport/spi/AbstractStreamClient.java
@@ -46,6 +46,14 @@ public abstract class AbstractStreamClient<C extends StreamClientConfiguration, 
     public StreamResponseMessage sendRequest(StreamRequestMessage requestMessage) throws InterruptedException {
         log.trace("Preparing HTTP request: " + requestMessage);
 
+        String[] split = requestMessage.getUri().toString().split(":");
+        String protocol = split[0];
+
+        if (protocol.equals("https")) {
+            log.debug("HTTPS invalid.  Ignoring call {}.",requestMessage.getUri());
+            return null;
+        }
+	
         // We want to track how long it takes
         long start = System.nanoTime();
 

